math/mathcore/inc/Fit/EvaluateLogL.hxx

```diff
@@ -0,0 +1,569 @@
+         if (!normalizeFunc) {
+            if (data.NDim() == 1) {
+               T x;
+               vecCore::Load<T>(x, data.GetCoordComponent(0, 0));
+               func( &x, p);
+            }
+            else {
+               std::vector<T> x(data.NDim());
+               for (unsigned int j = 0; j < data.NDim(); ++j)
+                  vecCore::Load<T>(x[j], data.GetCoordComponent(0, j));
+               func( x.data(), p);
+            }
+         }
+#endif
+
+         // this is needed if function must be normalized
+         double norm = 1.0;
+         if (normalizeFunc) {
+            // compute integral of the function
+            std::vector<double> xmin(data.NDim());
+            std::vector<double> xmax(data.NDim());
+            IntegralEvaluator<IModelFunctionTempl<T>> igEval(func, p, true);
+            // compute integral in the ranges where is defined
+            if (data.Range().Size() > 0) {
+               norm = 0;
+               for (unsigned int ir = 0; ir < data.Range().Size(); ++ir) {
+                  data.Range().GetRange(&xmin[0], &xmax[0], ir);
+                  norm += igEval.Integral(xmin.data(), xmax.data());
+               }
+            } else {
+               // use (-inf +inf)
+               data.Range().GetRange(&xmin[0], &xmax[0]);
+               // check if funcition is zero at +- inf
+               T xmin_v, xmax_v;
+               vecCore::Load<T>(xmin_v, xmin.data());
+               vecCore::Load<T>(xmax_v, xmax.data());
+               if (vecCore::ReduceAdd(func(&xmin_v, p)) != 0 || vecCore::ReduceAdd(func(&xmax_v, p)) != 0) {
+                  MATH_ERROR_MSG("FitUtil::EvaluateLogLikelihood", "A range has not been set and the function is not zero at +/- inf");
+                  return 0;
+               }
+               norm = igEval.Integral(&xmin[0], &xmax[0]);
+            }
+         }
+
+         // needed to compute effective global weight in case of extended likelihood
+
+         auto vecSize = vecCore::VectorSize<T>();
+         unsigned int numVectors = n / vecSize;
+
+         auto mapFunction = [ &, p](const unsigned i) {
+            T W{};
+            T W2{};
+            T fval{};
+
+            (void)p; /* avoid unused lambda capture warning if PARAMCACHE is disabled */
+
+            T x1;
+            vecCore::Load<T>(x1, data.GetCoordComponent(i * vecSize, 0));
+            const T *x = nullptr;
+            unsigned int ndim = data.NDim();
+            std::vector<T> xc;
+            if (ndim > 1) {
+               xc.resize(ndim);
+               xc[0] = x1;
+               for (unsigned int j = 1; j < ndim; ++j)
+                  vecCore::Load<T>(xc[j], data.GetCoordComponent(i * vecSize, j));
+               x = xc.data();
+            } else {
+               x = &x1;
+            }
+
+#ifdef USE_PARAMCACHE
+            fval = func(x);
+#else
+            fval = func(x, p);
+#endif
```

Comment:
Why is there this ifdef? Is it tested both ways? It looks like something that may stop working without we realizing it and then half of the code within ifdefs will be effectively dead code. Can we choose just to either enable or disable this and delete the unused code?

Reply:
@lmoneta ? I don't like that much the paramcache define either.